tutorials/truffle-dev-life-cycle.md

```diff
@@ -0,0 +1,538 @@
+
+```
+npx truffle test --network dev
+```
+
+!!! challenge
+    Try adding a test that uses a `tokenUri` for an NFT that hasn't approved the `NftMarketplace` contract to transfer it. You should assert that the call will fail.
+
+## Deploy to a Local Development Node {: #deploy-to-local-dev-node }
+
+Now that we feel confident in our NFT marketplace, we can go ahead and deploy our contracts. First, we'll need to update the deployment migration so that we deploy the `NftMarketplace` contract followed by the `DizzyDragons` contract, as we'll need to pass in the address of the marketplace to the constructor of the `DizzyDragons` contract. 
+
+To update the deployment script, open up the `migrations/2_deploy_contracts.js` migration file and replace it with the following:
+
+```js
+var NftMarketplace = artifacts.require("NftMarketplace");
+var DizzyDragons = artifacts.require("DizzyDragons");
+
+module.exports = async function (deployer) {
+  // Deploy the NFT Marketplace
+  await deployer.deploy(NftMarketplace);
+  const nftMarketplace = await NftMarketplace.deployed();
+  // Deploy the Dizzy Dragons contract
+  await deployer.deploy(DizzyDragons, nftMarketplace.address);
+};
+```
+
+Now we're all set to deploy our contracts. To do deploy our contracts to our local development node, run this command:
+
+```
+npx truffle migrate --network dev
+```
+
+You should see the transaction hashes for the deployment of each contract in your terminal. With our contracts deployed, we could begin to build a dApp with a frontend that interacts with these contracts, but it's out of scope for this tutorial.
+
+Now that we're done testing on the Moonbeam development node, don't forget to stop and remove the node! You can do so by running:
+
+```
+npx truffle run moonbeam stop && \
+npx truffle run moonbeam remove
+```
+
+## Deploying to Moonbase Alpha TestNet {: #deploying-to-moonbase-alpha }
+
+Now that we've been able to rapidly develop our contracts with our Moonbeam development node and feel confident with our code, we can move on to testing it on the Moonbase Alpha TestNet. 
+
+First, you'll need to update your `truffle-config.js` file and add in the private key of your account on Moonbase Alpha. The `privateKeyMoonbase` variable already exists, you just need to set it to your private key. **This is just for demonstration purposes only, never store your private keys in a JavaScript file**.
+
+Once you've set your account up, you can run your tests on Moonbase Alpha to make sure they work as expected on a live network:
+
+```
+npx truffle test --network moonbase
+```
+
+If your tests all pass, you can feel free to deploy your contracts using this command:
+
+```
+npx truffle migrate --network moonbase
+```
+
+Feel free to play around and interact with your contracts on the TestNet! Since DEV tokens have no real monetary value, now is a good time to work out any kinks before we deploy our contracts to Moonbeam MainNet where the tokens do have value!
```

Comment:
You can add the extra step of verifying the contract also in Moonscan Moonbase, and then Moonbeam as well. And provide an URL of the contract on Moonscan so people can check it out